runelite-client/src/main/java/net/runelite/client/plugins/highalchemy/HighAlchemyOverlay.java

```diff
@@ -0,0 +1,138 @@
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package net.runelite.client.plugins.highalchemy;
+
+import java.awt.Color;
+import java.awt.Graphics2D;
+import java.awt.Rectangle;
+import java.awt.image.BufferedImage;
+import java.util.HashSet;
+import java.util.Set;
+import javax.inject.Inject;
+import net.runelite.api.ItemComposition;
+import net.runelite.api.ItemID;
+import net.runelite.api.widgets.Widget;
+import static net.runelite.api.widgets.WidgetInfo.TO_GROUP;
+import net.runelite.api.widgets.WidgetItem;
+import net.runelite.client.game.ItemManager;
+import net.runelite.client.ui.overlay.WidgetItemOverlay;
+
+public class HighAlchemyOverlay extends WidgetItemOverlay
+{
+	private final ItemManager itemManager;
+	private final HighAlchemyConfig config;
+	private final HighAlchemyPlugin plugin;
+	private static final float HIGH_ALCHEMY_CONSTANT = 0.6f;
+
+	@Inject
+	public HighAlchemyOverlay(ItemManager itemManager, HighAlchemyPlugin plugin, HighAlchemyConfig config)
+	{
+		this.itemManager = itemManager;
+		this.config = config;
+		this.plugin = plugin;
+
+		showOnBank();
+		showOnInventory();
+	}
+
+	@Override
+	public void renderItemOverlay(Graphics2D graphics, int itemId, WidgetItem itemWidget)
+	{
+		Widget widget = itemWidget.getWidget();
+		int interfaceGroup = TO_GROUP(widget.getId());
+
+		if (!plugin.getInterfaceGroups().contains(interfaceGroup))
+		{
+			return;
+		}
+
+		final int id = getNotedId(itemId);
+		final int gePrice = getGEPrice(id);
+		final int haPrice = getHAPrice(id);
+		final int materialCost = getAlchCost();
+		final int desiredProfit = config.minProfit();
+		final int haProfit = getHAProfit(haPrice, gePrice, materialCost);
+
+		if (gePrice > 0 && haPrice > 0 && haProfit >= desiredProfit)
+		{
+			final Color color = config.getHighlightColor();
+
+			if (color != null)
+			{
+				Rectangle bounds = itemWidget.getCanvasBounds();
+				final BufferedImage outline = itemManager.getItemOutline(itemId, itemWidget.getQuantity(), color);
+				graphics.drawImage(outline, (int) bounds.getX() + 1, (int) bounds.getY() + 1, null);
+			}
+		}
+	}
+
+	private int getGEPrice(int id)
+	{
+		return itemManager.getItemPrice(id);
+	}
+
+	private int getHAPrice(int id)
+	{
+		if (id == ItemID.COINS_995)
+		{
+			return 0;
+		}
+		ItemComposition itemComp = itemManager.getItemComposition(id);
+		float haValue = itemComp.getPrice() * HIGH_ALCHEMY_CONSTANT;
+		return Math.round(haValue);
+	}
+
+	private int getHAProfit(int haPrice, int gePrice, int alchCost)
+	{
+		return haPrice - gePrice - alchCost;
+	}
+
+	private int getAlchCost()
+	{
+		// int natureRunePrice = itemManager.getItemPrice(ItemID.NATURE_RUNE);
+		int natureRunePrice = 0;
+		if (config.usingFireRunes())
+		{
+			int fireRunePrice = itemManager.getItemPrice(ItemID.FIRE_RUNE) * 5;
```

Comment:
Lava/steam/smoke runes? I personally never use fire runes as lavas are cheaper, may be better to check inventory/rune pouch for that

Reply:
Only really useful for slayer, which you dont care about profit as youll alch drops anyway from things such as gargoyles